feat/xchain custom gas payer

alber70g opened the pull request:
- chore(xchain): moved xchain.js to xchain.js 
- refactor: create one state interface 
- feat(xchain): get SigData to sign in Chainweaver 
- chore(xchain): formatting 
- fix(xchain): remove double elements. Fix isSingleSig 
- feat(xchain): adds clist to keys 
- feat(xchain): adds class Keys to add pubkey + clist to array 
- feat(xchain): manually signed transaction can now be send to network 
- feat(xchain): clean ui. Add local call before sending. Fix loading SigData 
- feat(xchain): run /local call on gas-station flow

jmininger commented on docs/xchain.html:
```diff
@@ -1,139 +1,236 @@
+          </div>
+
+          <div id="signed-transaction-wrapper" class="ui field hidden">
+            <label class="">Signed Transaction</label>
```
Maybe `Signed Transaction (JSON)` instead to let the user know that they are using the `JSON` tab of sigbuilder results instead of the `CommandJSON` tab

alber70g replied:
☑️

ggobugi27 commented:
The error message from `getProof()` `Initial transfer is not confirmed yet. Please wait and try again.` doesn't seem to render  when the initial transaction is confirmed and not yet ready to give back proof.

jmininger commented on docs/xchain.js:
```diff
@@ -388,60 +605,333 @@ function validatePact() {
+  }
+}
+
+function isSingleSig(res) {
```
This doesn't work for multisigs that only require 1 sig. But maybe that's ok. Perhaps add a comment

alber70g commented:
> The error message from `getProof()` `Initial transfer is not confirmed yet. Please wait and try again.` doesn't seem to render when the initial transaction is confirmed and not yet ready to give back proof.

I didn't change that one I think... I see it's thrown as a message, not as an Error. Not familiar with what that does.

alber70g commented:
> > The error message from `getProof()` `Initial transfer is not confirmed yet. Please wait and try again.` doesn't seem to render when the initial transaction is confirmed and not yet ready to give back proof.
> 
> I didn't change that one I think... I see it's thrown as a message, not as an Error. Not familiar with what that does.

Fixed this error.

And added a /local call on the gas-station flow. It now shows when it's already finished:
`resumePact: pact completed`

ggobugi27 commented:
> > The error message from `getProof()` `Initial transfer is not confirmed yet. Please wait and try again.` doesn't seem to render when the initial transaction is confirmed and not yet ready to give back proof.
> 
> I didn't change that one I think... I see it's thrown as a message, not as an Error. Not familiar with what that does.

The message will pop up when proof is not ready to be fetched from `/spv` endpoint. (need several blocks for proof to be ready after initial transaction is completed). Specifically, it will receive error message`SPV target not reachable: target chain not reachable. Chainweb instance is too young` from `/spv` instead of the proof. 

Can we make it so that SigData is not rendered, and button is disabled in the case `getProof()` fails? This is important because otherwise, the command payload in SigData will contain null as proof, and the finishing tx will fail. 

You can test this case by starting transaction, wait for it to complete, and then right after that trying to finish cross-chain-transfer.
